src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -206,12 +206,30 @@ private static int ComputeSymbolMatchPriority(ISymbol symbol)
         var name = SymbolCompletionItem.GetSymbolName(item);
         var kind = SymbolCompletionItem.GetKind(item);
         var isGeneric = SymbolCompletionItem.GetSymbolIsGeneric(item);
-        var relatedDocumentIds = document.Project.Solution.GetRelatedDocumentIds(document.Id);
         var typeConvertibilityCache = new Dictionary<ITypeSymbol, bool>(SymbolEqualityComparer.Default);
 
+        // First try with the document we're currently within.
+        var description = await TryGetDescriptionAsync(document.Id).ConfigureAwait(false);
+        if (description != null)
+            return description;
```

Comment:
completion tests were depending on the undefined ordering of GetRelatedDocumentIds.  It worked there because we'd always find earlier added projects vs later added ones.  But that's not at all something guaranteed as projects mutate, so the product was busted, and tests weren't properly validating things.  Updated the product code to have the logic we really want (check current doc first, tehn related docs).